pkg/assembler/backends/ent/backend/sbom.go

```diff
@@ -373,6 +373,214 @@ func getSBOMObjectWithIncludes(q *ent.BillOfMaterialsQuery) *ent.BillOfMaterials
+		go func(ctx context.Context, b *EntBackend, sbomID string, first int, occursChan chan<- occurResult) {
+			var afterCursor *entgql.Cursor[uuid.UUID]
+			defer close(occursChan)
+			for {
+				occurQuery := b.client.Occurrence.Query().
+					Where(occurrence.HasIncludedInSbomsWith([]predicate.BillOfMaterials{
+						optionalPredicate(&sbomID, IDEQ)}...))
+
+				occurConnect, err := getOccurrenceObjectWithoutEdges(occurQuery).
+					Paginate(ctx, afterCursor, &first, nil, nil)
+				if err != nil {
+					occursChan <- occurResult{occurs: nil,
+						occurErr: fmt.Errorf("failed included occurrence query for hasSBOM with error: %w", err)}
+				}
+
+				var paginatedOccurs []*ent.Occurrence
+
+				for _, edge := range occurConnect.Edges {
+					paginatedOccurs = append(paginatedOccurs, edge.Node)
+				}
+
+				occursChan <- occurResult{occurs: paginatedOccurs,
+					occurErr: nil}
+
+				if !occurConnect.PageInfo.HasNextPage {
+					break
+				}
+				afterCursor = occurConnect.PageInfo.EndCursor
+			}
+		}(ctx, b, sbomID, includedFirst, occursChan)
+
+		for occur := range occursChan {
+			if occur.occurErr != nil {
+				return nil, fmt.Errorf("occurrence channel failure: %w", occur.occurErr)
+			}
+			includedOccurs = append(includedOccurs, occur.occurs...)
+		}
+
+		for dep := range depsChan {
+			if dep.depErr != nil {
+				return nil, fmt.Errorf("dependency channel failure: %w", dep.depErr)
+			}
+			includedDeps = append(includedDeps, dep.deps...)
+		}
+		reconstructedSBOM := toModelHasSBOMWithJustIncludedIDs(foundSBOM.Node, includedDeps, includedOccurs)
+		reconstructedSBOMs[sbomID] = reconstructedSBOM
+	}
+
+	var edges []*model.HasSBOMEdge
+	for id, edge := range reconstructedSBOMs {
+		edges = append(edges, &model.HasSBOMEdge{
+			Cursor: hasSBOMGlobalID(id),
+			Node:   edge,
+		})
+	}
+
+	if hasSBOMConnection.PageInfo.StartCursor != nil {
+		return &model.HasSBOMConnection{
+			TotalCount: hasSBOMConnection.TotalCount,
+			PageInfo: &model.PageInfo{
+				HasNextPage: hasSBOMConnection.PageInfo.HasNextPage,
+				StartCursor: ptrfrom.String(hasSBOMGlobalID(hasSBOMConnection.PageInfo.StartCursor.ID.String())),
+				EndCursor:   ptrfrom.String(hasSBOMGlobalID(hasSBOMConnection.PageInfo.EndCursor.ID.String())),
+			},
+			Edges: edges,
+		}, nil
+	} else {
+		// if not found return nil
+		return nil, nil
+	}
+}
+
+func (b *EntBackend) deleteHasSbom(ctx context.Context, hasSBOMID uuid.UUID) (bool, error) {
+	_, txErr := WithinTX(ctx, b.client, func(ctx context.Context) (*string, error) {
+		tx := ent.TxFromContext(ctx)
+
+		first := 1
+		hasSBOMIDString := hasSBOMID.String()
+		hasSBOMConnection, err := b.getHasSBOMForDeletion(ctx, &hasSBOMIDString, nil, &first)
+		if err != nil {
+			return nil, errors.Wrap(err, "failed to query for hasSBOM node for deletion with error")
+		}
+
+		// first delete isDependency and isOccurrence nodes that are part of the hasSBOM node
+		for _, hasSBOMNEdge := range hasSBOMConnection.Edges {
```

Comment:
I've been testing locally and it happen to me to have `hasSBOMConnection` to be `nil`
```suggestion
		if hasSBOMConnection != nil {
		    for _, hasSBOMNEdge := range hasSBOMConnection.Edges {
```

Reply:
Interesting, how were you testing this? Did you pass in an ID that does not exist?

Reply:
I've ingested two CycloneDX SBOMs changing just the `serialNumber` and the component's `version`  but keeping the same `components` array so the same `dependencies`.
Then I tried to delete the first one and it failed with the error reported above.

Reply:
Let me test this out and I will make the changes.

Reply:
So I found the issue and improved the deletion. Thanks @mrizzi! Please test again.